transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -572,21 +568,16 @@ private void processSelectedKeysOptimized(SelectionKey[] selectedKeys) {
             if (needsToSelectAgain) {
                 // null out entries in the array to allow to have it GC'ed once the Channel close
                 // See https://github.com/netty/netty/issues/2363
-                for (;;) {
-                    i++;
-                    if (selectedKeys[i] == null) {
-                        break;
-                    }
-                    selectedKeys[i] = null;
-                }
+                this.selectedKeys.reset(i + 1);
```

Comment:
The current code and the previous code should be logically equivalent but have the following differences:
- the previous implementation didn't utilize the size of the set when iterating. We now have a fixed/known boundary to iterate to (`size`) instead of conditional based on the next element contents.
- This set doesn't allow `null` elements to be added, so checking each element for `null` is not necessary.

I pushed some updated code to make the relationship with `this.selectedKeys` more clear.